AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/justice-user-form/justice-user-form.component.ts

```diff
@@ -47,21 +51,37 @@ export class JusticeUserFormComponent {
         });
     }
 
+    ngOnChanges(changes: SimpleChanges): void {
+        const mode = changes['mode'];
+        if (mode.currentValue === 'edit') {
+            ['firstName', 'lastName', 'username', 'contactTelephone'].forEach(field => this.form.controls[field].disable());
+        }
+    }
+
     onSave() {
         this.failedSaveMessage = null;
         this.showSpinner = true;
-        this.justiceUserService
-            .addNewJusticeUser(
-                this.form.controls.username.value,
-                this.form.controls.firstName.value,
-                this.form.controls.lastName.value,
-                this.form.controls.contactTelephone.value,
-                this.form.value.role
-            )
-            .subscribe({
-                next: newJusticeUser => this.onSaveSucceeded(newJusticeUser),
-                error: (error: string | BookHearingException) => this.onSaveFailed(error)
-            });
+        if (this.mode === 'add') {
```

Comment:
Sorry to be pedantic, but would you mind breaking the blocks into their own methods so the onSave can be leaner and more verbose

```
onSave() {
 // code excluded
 if (this.mode === 'add') {
   this.addNewUser()
 } 
 else if (this.mode === 'edit') {
      this.updateExistingUser()
 }
}
```

Reply:
Done, thanks